eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php

```diff
@@ -349,8 +367,28 @@ public function publish( $typeId )
         // Clear type cache and all identifier cache (as we don't know the identifier)
         $this->cache->clear( 'contentType', $typeId );
         $this->cache->clear( 'contentType', 'identifier' );
+        $this->cache->clear( 'fieldMap' );
 
         // clear content cache
         $this->cache->clear( 'content' );//TIMBER! (possible content changes)
     }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\Type\Handler::getFieldMap
+     */
+    public function getFieldMap( $legacy = false )
```

Comment:
ok 

Ideally places that need to get the legacy representation should be adapted, structure should be richer, or they should use different service.
But if that is not possible at least add some more doc on it explaining what it does, 'legacy' does not tell anything regarding change in structure, that it is temporary and that is not part of interface by intention.